File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int m = INT_MAX;
        for(int price : prices) {
            ans = max(ans, price - m);
            m = min(m, price);
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int sol = 0, min = Integer.MAX_VALUE;
        
        for(int i=0; i<prices.length; i++){
            min = Math.min(min, prices[i]);
            sol = Math.max(prices[i] - min, sol);
        }
        
        return sol;
    }
}